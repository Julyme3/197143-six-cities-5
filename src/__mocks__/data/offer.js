/* eslint-disable */
export const offersFromAPi = [
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 1,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 2,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
];

export const offersAdapteToClient = [
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 2,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
];

export const offerFromApi = {
  bedrooms: 3,
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13,
    },
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
  host: {
    id: 25,
    name: `Angelina`,
    is_pro: true,
    avatar_url: `img/avatar-angelina.jpg`,
  },
  id: 1,
  images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
  is_favorite: false,
  is_premium: false,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16,
  },
  max_adults: 4,
  preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
  price: 493,
  rating: 3.8,
  title: `The house among olive`,
  type: `house`,
};

export const fullOfferAdaptedToClient = {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  };

export const commentsFromApi = [
  {
    comment: "The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.",
    date: "2020-11-05T13:38:44.753Z",
    id: 1,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg",
      id: 16,
      is_pro: true,
      name: "Mollie",
    },
  },
  {
    comment: "I stayed here for one night and it was an unpleasant experience.",
    date: "2020-11-05T13:38:44.753Z",
    id: 2,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
      id: 19,
      is_pro: false,
      name: "Christina",
    },
  },
];

export const commentsAdaptedToCLient = [
  {
    id: 3,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
    name: "Isaac",
    raiting: 5,
    text: "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
    dateFormated: "November 2020",
  },
  {
    id: 2,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
    name: "Christina",
    raiting: 4,
    text: "I stayed here for one night and it was an unpleasant experience.",
    dateFormated: "November 2020",
  },
];

export const favoriteOffers = {
  "Amsterdam": [
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 1,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 2,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
  ],
};
