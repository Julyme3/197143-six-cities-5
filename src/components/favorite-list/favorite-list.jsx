import React, {useCallback} from "react";
import PropTypes from "prop-types";
import OfferCardFavorite from "../offer-card-favorite/offer-card-favorite";
import {postFavoriteAction} from "../../store/api-actions";
import {connect} from "react-redux";
import {updateFavoriteInListAction} from "../../store/actions";
import {OffersType} from "../../types";

const FavoriteList = (props) => {
  const {list, updateFavorite, postFavorite, onChangeActiveItem} = props;

  const handleChangeFavorite = useCallback((id, isFavorite) => {
    postFavorite(id, isFavorite, updateFavorite);
  }, [list]);

  return (
    list.length ? <div className="favorites__places">
      {list.map((offer) => {
        return (
          <OfferCardFavorite
            key={offer.id}
            offer={offer}
            onChangeFavorite={handleChangeFavorite}
            onChangeActiveOffer={onChangeActiveItem}
          />
        );
      })}
    </div>
      : null
  );
};

FavoriteList.propTypes = {
  list: OffersType,
  postFavorite: PropTypes.func.isRequired,
  updateFavorite: PropTypes.func.isRequired,
  onChangeActiveItem: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  postFavorite(id, status, action) {
    dispatch(postFavoriteAction(id, status, action));
  },
  updateFavorite(data) {
    dispatch(updateFavoriteInListAction(data));
  },
});

export {FavoriteList};
export default connect(null, mapDispatchToProps)(FavoriteList);
