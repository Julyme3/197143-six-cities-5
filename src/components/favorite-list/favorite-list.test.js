import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import {FavoriteList} from "./favorite-list";
import {offersAdapteToClient as list} from "../../__mocks__/data/offer";

const noop = () => {};

describe(`FavoriteList render`, () => {
  it(`FavoriteList with items`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FavoriteList
              list={list}
              postFavorite={noop}
              updateFavorite={noop}
              onChangeActiveItem={noop}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoriteList without items`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FavoriteList
              list={[]}
              postFavorite={noop}
              updateFavorite={noop}
              onChangeActiveItem={noop}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
