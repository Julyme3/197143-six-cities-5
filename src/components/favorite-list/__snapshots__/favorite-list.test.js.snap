// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList render FavoriteList with items 1`] = `
<div
  className="favorites__places"
>
  <article
    className=" favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=" favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="110px"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
          width="150px"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button
              false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className=" favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className=" favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="110px"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
          width="150px"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            493
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button
              false"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "76%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          The house among olive 
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;

exports[`FavoriteList render FavoriteList without items 1`] = `null`;
